core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala

```diff
@@ -142,9 +142,11 @@ private class AsyncEventQueue(
       eventCount.incrementAndGet()
       eventQueue.put(POISON_PILL)
     }
-    // this thread might be trying to stop itself as part of error handling -- we can't join
+    // 1.If the user does not want to wait for the dispatch to end,
+    // they can omit the thread join.
+    // 2.this thread might be trying to stop itself as part of error handling -- we can't join
     // in that case.
-    if (Thread.currentThread() != dispatchThread) {
+    if (waitForEventDispatchExit() && Thread.currentThread() != dispatchThread) {
```

Comment:
Shall we remove `waitForEventDispatchExit()` and inline `conf.get(LISTENER_BUS_EVENT_QUEUE_WAIT_FOR_EVENT_DISPATCH_EXIT_ON_STOP)` ?

Reply:
I think use waitForEventDispatchExit is more readable here, it's explaining what it does. Using LISTENER_BUS_EVENT_QUEUE_WAIT_FOR_EVENT_DISPATCH_EXIT_ON_STOP, you have to jump to the place that define this conf to figure out the meaning.

Reply:
@beliefer Sure, already commit, please have a check